packages/xstate-solid/test/useMachine.test.tsx

```diff
@@ -232,6 +232,44 @@ describe('useMachine hook', () => {
     waitFor(() => screen.getByTestId('success')).then(() => done());
   });
 
+  it('initial send should work only with createRenderEffect in useMachine', (done) => {
+    const machine = createMachine({
+      initial: 'start',
+      states: {
+        start: {
+          on: {
+            done: 'success'
+          }
+        },
+        success: {
+          type: 'final'
+        }
+      }
+    });
+
+    const Spawner = () => {
+      const [current, send] = useMachine(machine);
+      // This should fail if useMachine is not using createRenderEffect
+      expect(current.value).toBe('start');
+      send({ type: 'done' });
```

Comment:
"Side effects during render" - my React brain now:
<img src="https://media.giphy.com/media/FcuiZUneg1YRAu1lH2/giphy.gif"/>

Reply:
I was able to remove the `createRenderEffect` in fsm that was tracking when a service was replaced. I can't do the same for `useActor` because `next` is not called on the initial `subscribe`. I can condense it down to a single `createRenderEffect` with the code below. Do you want to move in this direction or stick with what we're doing now?

```
  createRenderEffect<boolean>((initialActor) => {
    const actor = actorMemo();
    if (!initialActor) {
      setState({
        snapshot: deriveServiceState(actor, getActorState())
      });
    }

    const { unsubscribe } = actor.subscribe({
      next: (emitted: unknown) =>
        setState({
          snapshot: deriveServiceState(actor, emitted)
        }),
      error: noop,
      complete: noop
    });
    onCleanup(unsubscribe);

    return false;
  }, true);
```

Reply:
Could also do this:

```typescript
  const setActorState = (actorState: unknown) => {
    setState({
      snapshot: deriveServiceState(actorMemo(), actorState)
    });
  };

  createRenderEffect<boolean>((isInitialActor) => {
    const actor = actorMemo();
    if (!isInitialActor) {
      setActorState(getActorState());
    }

    const { unsubscribe } = actor.subscribe({
      next: setActorState,
      error: noop,
      complete: noop
    });
    onCleanup(unsubscribe);

    return false;
  }, true);
```

Reply:
One other thing to consider, replacing the current actor/service only works with `createComputed` in place of `createRenderEffect`. I know the `createComputed` is generally not recommended but our use case may call for it. 

```typescript
  const [service, setService] = createSignal(interpret(machine).start());
  const [current, send] = useActor(service);

  expect(current().value).toBe('start');
  send({ type: 'done' });
  expect(current().value).toBe('success');

  // replace the current service, only works with 'createComputed'
  setService(interpret(machine).start());
  expect(current().value).toBe('start');
  send({ type: 'done' });
```

Reply:
After reviewing and thinking on this more, `createComputed` fits our use case in this situation. It prevents the state returned from `useActor` from being out of sync with actor.getSnapshot().